app/src/services/mongo/MongoDBConnection.ts

```diff
@@ -0,0 +1,88 @@
+import mongoose, {Mongoose} from 'mongoose';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {getLogger} from '../../logger';
+import {injectable} from 'inversify';
+import Database, {DatabaseConfig} from '../../config/database';
+
+@injectable()
+export default class MongoDBConnection {
+    private static readonly logger = getLogger('DatabaseConnection');
+    private static isConnected = false;
+    private static connecting = false;
+    private static db: Mongoose;
+    private static dbConfig: Database;
+
+    constructor(dbConfig: Database) {
+        MongoDBConnection.dbConfig = dbConfig;
+    }
+
+    public getConnection(): Mongoose|boolean {
+        if (MongoDBConnection.connecting) {
+            MongoDBConnection.logger.info('MongoDBConnection is currently connecting, aborting connection.');
+            return false;
+        }
+
+        if (MongoDBConnection.isConnected) {
+            return MongoDBConnection.db;
+        } else {
+            try {
+                void MongoDBConnection.connect();
```

Comment:
Void? I think this should also be await

Reply:
Have to convert parent function to async which causes issues, think the method needs rewriting